GitHub_Learning/Feature/Onboarding/View/OnboardingViewController.swift

```diff
@@ -28,5 +33,71 @@ class OnboardingViewController: BaseViewController {
     
     override func viewDidLoad() {
         super.viewDidLoad()
+        layout.pagesCollectionView.dataSource = self
+        layout.pagesCollectionView.delegate = self
+        layout.nextButton.addTarget(self, action: #selector(nextButtonPressed), for: .touchUpInside)
+        presenter.viewDidLoad()
+    }
+    
+    @objc private func nextButtonPressed(sender: UIButton) {
+        let nextPage = Int(layout.pagesCollectionView.contentOffset.x/view.frame.width) + 1
+        presenter.updatePageControlFor(nextPage)
+        presenter.updatePageTo(nextPage)
+        presenter.updateButtonTitleFor(nextPage)
+    }
+    
+}
+
+extension OnboardingViewController: UICollectionViewDataSource, UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        onboardingPagesVM.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: OnboardingCell.onboardingCellId, for: indexPath) as! OnboardingCell
+        let item = onboardingPagesVM[indexPath.item]
+        cell.onboardingPageViewModel = item
+        return cell
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return CGSize(width: view.bounds.width, height: layout.pagesCollectionView.frame.height)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+        return 0
     }
+    
+    func scrollViewDidEndDecelerating(_ scrollView: UIScrollView) {
+        let page = Int(scrollView.contentOffset.x/view.frame.width)
```

Comment:
Toto je logika, ktera by mela byt v presenteru. Navic, ty to tady spocitas a pak jen reknes presenteru, aby ti rek, at to nastavis. To je k nicemu.

Reply:
S těma moc nevím jak to bych to tahal pro presenteru, kdybych zachoval stejný funkce který jsi navrhoval níž. + logika na počítání toho page je i ve funkci nextButtonPressed, tam myslíš to taky vadí?